test/parallel/test-fs-cp-sync-dereference-true.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+// Refs: https://github.com/nodejs/node/issues/58939
+//
+// In this test, both the cp and cpSync functions are attempting to copy
+// a file over a symlinked directory.
```

Comment:
Unlike the original `test-fs-cp-sync-dereference.js`, this one seems to test only the synchronous version.